### PR TITLE
Updated homepage, installation, and usage docs

### PR DESCRIPTION
**Pull request by ellbosch**
Tag: #310 
> Note: Git's diff is largely ineffective for this PR. I've provided 'before and after' links to showcase doc changes.

This PR introduces sweeping changes to our docs in preparation for GA. Changes include:
* New README.md for homepage ([before](https://github.com/dbcli/mssql-cli/blob/d5c24a2a1846b094c8a21befbf46dd1c33eee3b6/README.rst) and [after](https://github.com/dbcli/mssql-cli/blob/ellbosch/install-docs/README.md))
    * Expanded description for features
    * New 'Quick Start' section to get users up and running faster
    * Updated python badge
* New README.md for docs/installation ([before](https://github.com/dbcli/mssql-cli/blob/d5c24a2a1846b094c8a21befbf46dd1c33eee3b6/doc/installation_guide.md) and [after](https://github.com/dbcli/mssql-cli/tree/ellbosch/install-docs/doc/installation)), replacing old installation_guide.md
* Improved Linux installation docs ([before](https://github.com/dbcli/mssql-cli/blob/d5c24a2a1846b094c8a21befbf46dd1c33eee3b6/doc/installation/linux.md) and [after](https://github.com/dbcli/mssql-cli/blob/ellbosch/install-docs/doc/installation/linux.md))
    * Expanded Ubuntu section per release
    * Removed repetitious language around direct downloads
* Improved usage docs ([before](https://github.com/dbcli/mssql-cli/blob/d5c24a2a1846b094c8a21befbf46dd1c33eee3b6/doc/usage_guide.md) and [after](https://github.com/dbcli/mssql-cli/blob/ellbosch/install-docs/doc/usage_guide.md)): added config file section, other minor changes
* Brings documentation back to our PyPI profile site

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+[![PyPI](https://badge.fury.io/py/mssql-cli.svg)](https://pypi.python.org/pypi/mssql-cli)
+[![Python 2,3](https://img.shields.io/badge/python-2.7,&nbsp;3.x-blue.svg)](https://github.com/dbcli/mssql-cli)
```
Comment (pensivebrian) [inline README.md]: I think we should be explicit about which python versions we support and avoid 3.x.

**Comment by ellbosch** [comment]
@pensivebrian sorry for muddling in some minor pep8 related fixes, I can separate those changes into a separate PR if you'd prefer that.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+Visit the [installation reference guide](https://github.com/dbcli/mssql-cli/tree/master/doc/installation) to view all supported releases and downloads.
+
+#### Install with Linux Package Manager
+> A new stable version of mssql-cli is coming March 2020 for `apt-get` and `yum`.
+
+Follow the [Linux installation instructions]('https://github.com/dbcli/mssql-cli/blob/master/doc/installation/linux.md') to install mssql-cli using `apt-get`, `yum`, and other Linux package managers.
+
+#### Install with pip
+```
+pip install mssql-cli
```
Comment (pensivebrian) [inline README.md]: Let's use: `python -m pip install mssql-cli` and remove all other references to running pip directly

File: README.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+### Install mssql-cli
+Platform-specific installation instructions are below:
+| [Windows](https://github.com/dbcli/mssql-cli/blob/master/doc/installation/windows.md#windows-installation) | [macOS](https://github.com/dbcli/mssql-cli/blob/master/doc/installation/macos.md#macos-installation) | [Linux](https://github.com/dbcli/mssql-cli/blob/master/doc/installation/linux.md) |
+| - | - | - |
+
+Visit the [installation reference guide](https://github.com/dbcli/mssql-cli/tree/master/doc/installation) to view all supported releases and downloads.
+
+#### Install with Linux Package Manager
+> A new stable version of mssql-cli is coming March 2020 for `apt-get` and `yum`.
```
Comment (pensivebrian) [inline README.md]: Let's remove this line

File: README.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+Follow the [Linux installation instructions]('https://github.com/dbcli/mssql-cli/blob/master/doc/installation/linux.md') to install mssql-cli using `apt-get`, `yum`, and other Linux package managers.
+
+#### Install with pip
+```
+pip install mssql-cli
+```
+Please refer to the [pip installation docs](https://github.com/dbcli/mssql-cli/blob/master/doc/installation/pip.md) for more platform-specific information.
+
+#### Install Daily Builds
```
Comment (pensivebrian) [inline README.md]: Let's move this to a daily build section - I want to keep the home page clean.

File: build.py [hunk 5]
```diff
@@ -126,9 +131,8 @@ def validate_package():
     # cache overshadowing actual changes made.
     current_platform = utility.get_current_platform()
 
-    mssqlcli_wheel_name = [
-        pkge for pkge in mssqlcli_wheel_dir if current_platform in pkge and
-        'dev-latest' not in pkge]
+    mssqlcli_wheel_name = [pkge for pkge in mssqlcli_wheel_dir
+                           if current_platform in pkge and 'dev-latest' not in pkge]
```
Comment (pensivebrian) [inline build.py]: I want to remove this dev-latest concept - it was a bad idea. But that's for another PR

File: doc/installation/README.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+| - | - | - |
+
+
+## Direct Downloads
+
+The instructions above are the preferred installation method. Direct downloads are provided as an alternative in the 
+scenario that your machine may not have access to the Microsoft package repository. Instructions for direct downloads can
+also be found in the links above.
+
+| Supported Platform                         |Latest Stable                 |Latest Dev               |
```
Comment (pensivebrian) [inline doc/installation/README.md]: Let's remove the latest dev column. If we ever make a fix that we want to make available, we can just point the user to the file in azure storage that has the fix.

File: doc/installation/linux.md [hunk 7]
```diff
@@ -1,66 +1,67 @@
...
-from the [downloads] page onto the Ubuntu machine.
+# Update the list of products
+sudo apt-get update
 
-Then execute the following in the terminal:
+# Install mssql-cli
+sudo apt-get install mssql-cli
+```
 
+##### Insiders Direct Download on Ubuntu
```
Comment (pensivebrian) [inline doc/installation/linux.md]: Again, let's remove all references to the direct download

**Pull request merged**